STYLE_GUIDE.md

```diff
@@ -0,0 +1,461 @@
+# Pony Standard Library Style Guide
+
+This is a document describing the preferred practices for writing code for the Pony standard library. Other projects are free to use the same style, but you are free to use any style in other projects. The following conventions are meant to ensure consistency and readability of Pony code. Any of the following rules may be broken in situations where the recommended practice would diminish readability.
+
+
+## Formatting
+
+### Line Length
+
+The maximum line length is 80 columns.
+
+### Trailing Whitespace
+
+No trailing whitespace at the end of lines.
+
+### Indentation
+
+Use 2 spaces for indentation, No hard tabs.
+
+```pony
+actor Main
+  new create(env: Env) =>
+    env.out.print("Hello, world!")
+```
+
+### Blank Lines
+
+- One blank line is placed between top-level declarations such as classes, primitives, actors, etc.
```

Comment:
I find that having the same one-line gap between functions and classes can cause the classes to get lost a bit. Especially since there is no `end` on functions and classes.